vyper/ast/nodes.py

```diff
@@ -15,34 +14,107 @@
 )
 
 BASE_NODE_ATTRIBUTES = (
-    'node_id',
-    'source_code',
+    'ast_type',
     'col_offset',
-    'lineno',
     'end_col_offset',
     'end_lineno',
-    'src'
+    'lineno',
+    'node_id',
+    'source_code',
+    'src',
 )
+DICT_AST_SKIPLIST = ('source_code', )
+
+
+def get_node(ast_struct: typing.Union[typing.Dict, python_ast.AST]) -> "VyperNode":
+    """
+    Converts an AST structure to a Vyper AST node.
+
+    This is a recursive call, all child nodes of the input value are also
+    converted to Vyper nodes.
+
+    Attributes
+    ----------
+    ast_struct: (dict, AST)
+        Annotated python AST node or Vyper AST dict to generate the node from.
+
+    Returns
+    -------
+    VyperNode
+        The generated AST object.
+    """
+    if not isinstance(ast_struct, dict):
+        ast_struct = ast_struct.__dict__
+
+    vy_class = getattr(sys.modules[__name__], ast_struct['ast_type'], None)
+
+    if vy_class is None:
+        raise SyntaxException(
+            f"Invalid syntax (unsupported '{ast_struct['ast_type']}'' Python AST node).",
+            ast_struct
+        )
+
+    return vy_class(**ast_struct)
+
+
+def _to_node(value):
+    if isinstance(value, (dict, python_ast.AST)):
+        return get_node(value)
+    return value
```

Comment:
Shouldn't this throw if conversion is not possible? Or do we not care because it's internal?

Reply:
This is me avoiding duplicated code, basically.  I pass _every_ member of the node through it, and if the value is a node, it gets converted.  Without the function I'd have to write some ugly code for evaluating items in lists.